### ISSUE TITLE
Validation Set for week 4

### ISSUE THREAD

**Issue by mengchantang** [dmc-2016/dmc#80]
By trying to reduce overfitting I have created a validation set and have created variables/functions identical to those of the training and test sets. I have also plotted the accuracy of test set over epochs and seems have reached a much better result than a model without the validation set. 

However, there is one thing that puzzles me, how do I know that the model is using the validation set to train the model? It seems that there is nowhere for me to replace the test set with validation set in the code.
The only lines that deals with calculating the cost are these:
`loss = tf.reduce_mean(tf.square(tf.sub(prediction, _y)))`
   `optimizer = tf.train.GradientDescentOptimizer(learning_rate).minimize(loss)`
They do not appear to be specifying which data to use for the calculation as they are defined to deal with the generic.
